### PR TITLE
feat: ask for user approval for API requests

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: ui/app/AppLayouts/Browser/BrowserLayout.qml [hunk 1]
```diff
@@ -24,14 +28,90 @@ Item {
...
     QtObject {
         id: provider
         WebChannel.id: "backend"
 
         signal web3Response(string data);
 
         function postMessage(data){
-            web3Response(web3Provider.postMessage(data));
+            request = JSON.parse(data)
+            if(request.type === "api-request"){
```
Comment (jrainville) [inline ui/app/AppLayouts/Browser/BrowserLayout.qml:108]: Can it be a constant?

**Review by jrainville — APPROVED** [review]
Nice job

**Review by jrainville — APPROVED** [review]

**Review by emizzle — COMMENTED** [review]
Where are these permissions stored (so we don't have to keep asking)? In `status-go`?

**Comment by richard-ramos** [comment]
> Where are these permissions stored (so we don't have to keep asking)? In `status-go`?

Yes. there's a setting where you store the permission and website that asked for it.

**Pull request merged**